packages/network-controller/src/NetworkController.ts

```diff
@@ -274,7 +270,12 @@ export class NetworkController extends BaseControllerV2<
         if (chainId === undefined) {
           throw new Error('chainId must be passed in for custom rpcs');
         }
-        rpcTarget && this.setupStandardProvider(rpcTarget, chainId);
+
+        if (rpcTarget === undefined) {
+          throw new Error('rpcTarget must be passed in for custom rpcs');
+        }
+        const cid: Hex = chainId.startsWith('0x') ? chainId as Hex : `0x${parseInt(chainId, 10).toString(16)}`;
```

Comment:
Is there a function from `@metamask/utils` we can use instead of this? `add0x` might help at least with getting TypeScript to "see" a string as a `Hex`. I think it'd be great to avoid the typecast if we can.

Reply:
The better fix would be to make the network controller's interface always use `Hex` for chainId, and possibly have it throw aswell (for non-typescript consumers).

That said, Ill change the conversion to use `metamask-utils#toHex`.